src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -119,7 +122,7 @@ public static class TagHelperBlockRewriter
                 else if (name == null && symbol.Type == HtmlSymbolType.Text)
                 {
                     name = symbol.Content;
```

Comment:
:+1: will add comments at the top of each branch stating where the cursor currently points and why.